### PR TITLE
stochatreat

### PR DESCRIPTION
**Pull request by RoyalTS**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/stochatreat/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: Please remove the explanatory comments.

File: recipes/stochatreat/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  # See https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python for more details.
+  # noarch: python
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: Remove the build section if not needed.

File: recipes/stochatreat/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    Mainly thought with RCTs in mind, it also works for any other scenario in
+    where you would like to randomly allocate treatment within blocks or strata.
+  # doc_url: http://stochatreat.readthedocs.io/
+  dev_url: https://github.com/manmartgarc/stochatreat
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
+    # Always check with the people listed below if they are OK becoming maintainers of the recipe. (There will be spam!)
+    - manmartgarc
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: @manmartgarc Are you ok with being a maintainer?

File: recipes/stochatreat/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  # Remember to specify the license variants for BSD, Apache, GPL, and LGPL.
+  # Prefer the short version, e.g: GPL-2.0 instead of GNU General Public License version 2.0
+  # See https://opensource.org/licenses/alphabetical
+  license: MIT
+  # The license_family, i.e. "BSD" if license is "BSD-3-Clause". (optional)
+  license_family: MIT
+  # It is strongly encouraged to include a license file in the package,
+  # (even if the license doesn't require it) using the license_file entry.
+  # See http://conda.pydata.org/docs/building/meta-yaml.html#license-file
+  license_file: LICENSE
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: Add the LICENSE file to the recipe until https://github.com/manmartgarc/stochatreat/pull/5 is merged.

File: recipes/stochatreat/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    # - {{ compiler('c') }}
+  host:
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: `setuptools` is missing

File: recipes/stochatreat/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    - python
+    - pip
+    - pandas
+    - numpy
+  run:
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: All the dependencies in `host` must be copied here for this package

File: recipes/stochatreat/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  # uncomment the line below, and modify as needed.
+  # url: https://github.com/stochatreat/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 3522942b038c6f8a7aaa0e792c15ef988428ef523485cb430d13b2277967ab20
+  # sha256 is the preferred checksum -- you can get it for a file with:
+  #  `openssl sha256 <file name>`.
+  # You may need the openssl package, available on conda-forge:
+  #  `conda install openssl -c conda-forge``
+
+build:
+  # Uncomment the following line if the package is pure Python and the recipe is exactly the same for all platforms.
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: As it doesn't have compiled code, this should be `noarch: python`

File: recipes/stochatreat/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    - pip
+    - setuptools
+    - typing
+    - pandas
+    - numpy
+
+test:
+  imports:
+    - stochatreat
+  source_files:
```
Comment (xhochy) [inline recipes/stochatreat/meta.yaml]: You need to add those to the recipe to run them but import tests are fine in conda packages, so just skip this.

**Comment by RoyalTS** [comment]
@xhochy Build currently fails because of the type hints in the function signature. I assume one would have to either get rid of those or require Python 3.5+? If so, how do I do that?

**Comment by RoyalTS** [comment]
Never mind, figured it out. Seems like everything is building now!

**Pull request merged**